radlab-ui/webapp/src/utils/mailHandler.ts

```diff
@@ -0,0 +1,186 @@
+
+  if (mailSubject === "RAD Lab Module has been deleted for you!") {
+    const html = await readFileSync(
+      path.join(configDirectory + "/assets/htmlTemplates/deleteEmail.html"),
+      "utf8",
+    )
+    const template = await handlebars.compile(html)
+
+    const data = {
+      deploymentId,
+      deploymentLink: `${server}/deployments/${deploymentId}`,
+    }
+
+    handlebars.registerHelper("deployment_link", () => {
+      return new handlebars.SafeString(
+        handlebars.Utils.escapeExpression(data.deploymentLink),
+      )
+    })
+
+    const htmlToSend = template(data)
+
+    const mailOptions: IEmailOptions = {
+      recipients,
+      subject: mailSubject,
+      mailBody: htmlToSend,
+      credentials: {
+        email: settings.variables.mail_server_email,
+        password,
+      },
+    }
+
+    await sendMail(mailOptions)
+  } else {
+    const html = await readFileSync(
+      path.join(configDirectory + "/assets/htmlTemplates/email.html"),
+      "utf8",
+    )
+
+    const template = await handlebars.compile(html)
+
+    const billingId = variables.billing_account_id as string
+    const maskedBillingId =
+      billingId.substring(0, billingId.length - 6).replace(/[a-z\d]/gi, "*") +
+      billingId.substring(billingId.length - 6, billingId.length)
+
+    const outputs = [
+      {
+        name: "module",
+        value: module,
+      },
+      {
+        name: "billing_account_id",
+        value: maskedBillingId,
+      },
+      {
+        name: "zone",
+        value: variables.zone,
+      },
+    ]
+
+    const data = {
+      outputs,
+      mailBodyTitle: mailSubject,
+      projectId,
+      projectLink: `https://console.cloud.google.com/welcome?project=${projectId}`,
+      deploymentId,
+      deploymentLink: `${server}/deployments/${deploymentId}`,
+    }
+    handlebars.registerHelper("gcp_project_link", () => {
+      return new handlebars.SafeString(
+        handlebars.Utils.escapeExpression(data.projectLink),
+      )
+    })
+
+    handlebars.registerHelper("deployment_link", () => {
+      return new handlebars.SafeString(
+        handlebars.Utils.escapeExpression(data.deploymentLink),
+      )
+    })
+
+    const htmlToSend = template(data)
+
+    const mailOptions: IEmailOptions = {
+      recipients,
+      subject: mailSubject,
+      mailBody: htmlToSend,
+      credentials: {
+        email: settings.variables.mail_server_email,
+        password,
+      },
+    }
+
+    await sendMail(mailOptions)
+  }
+}
+
+export const sendMail = async (emailOptions: IEmailOptions) => {
+  const { recipients, credentials } = emailOptions
+  if (!recipients.length) {
+    return
+  }
+
+  const uniqueRecipients = [...new Set(recipients)]
+
+  let transporter = nodemailer.createTransport({
+    service: "gmail",
+    auth: {
+      user: credentials.email,
+      pass: credentials.password,
+    },
+  })
+
+  transporter.verify().then(console.log).catch(console.error)
```

Comment:
I think you need to `await` this. And do we need to `console.log` on success?

Reply:
Updated to `await` and removed `then` block